### PR TITLE
Adding ORDER support for UNION queries

### PR DESCRIPTION
**Pull request by mhtghn**

### CODE REVIEW

File: src/QueryFactory/FindObjectsFromRawSqlQueryFactory.php [hunk 1]
```diff
@@ -130,12 +132,46 @@ private function processParsedUnionQuery(array $parsedSql, ?string $sqlCount): a
...
+                $first = false;
+            }
+
+            $sql .= $select_builder->build($clause);
+        }
+
+        // Close the parenthesis if needed
+        if (!$first) {
+            $sql .= ")";
+        }
```
Comment (homersimpsons) [inline src/QueryFactory/FindObjectsFromRawSqlQueryFactory.php:170]: Can this be updated to a `array_map` and `implode` call ?

File: src/QueryFactory/FindObjectsFromRawSqlQueryFactory.php [hunk 2]
```diff
@@ -130,12 +132,46 @@ private function processParsedUnionQuery(array $parsedSql, ?string $sqlCount): a
 
         $generator = new PHPSQLCreator();
```
Comment (homersimpsons) [inline src/QueryFactory/FindObjectsFromRawSqlQueryFactory.php:133]: Is it still necesarry ? could this still be used instead of the custom `buildUnion` ?

**Review by homersimpsons — CHANGES_REQUESTED** [review]

File: src/QueryFactory/FindObjectsFromRawSqlQueryFactory.php (hunk 1 above)
Comment (mhtghn) [reply src/QueryFactory/FindObjectsFromRawSqlQueryFactory.php]: done

File: src/QueryFactory/FindObjectsFromRawSqlQueryFactory.php (hunk 2 above)
Comment (mhtghn) [reply src/QueryFactory/FindObjectsFromRawSqlQueryFactory.php:133]: yes the generator is used to create the processed sql count query just below the `buildUnion` call

`$processedSqlCount = $generator->create($countQuery);`

File: src/QueryFactory/FindObjectsFromRawSqlQueryFactory.php [hunk 3]
```diff
@@ -130,12 +132,30 @@ private function processParsedUnionQuery(array $parsedSql, ?string $sqlCount): a
...
         return [$processedSql, $sqlCount ?? $processedSqlCount, $columnDescriptors];
     }
 
+    /**
+     * @param mixed[] $parsed
+     */
+    private function buildUnion(array $parsed): string
+    {
+        return implode(' UNION ', array_map(function ($clause) {
+            $select_builder = new SelectStatementBuilder();
```
Comment (homersimpsons) [inline src/QueryFactory/FindObjectsFromRawSqlQueryFactory.php:154]: Use `camelCase`

**Review by homersimpsons — CHANGES_REQUESTED** [review]

File: src/QueryFactory/FindObjectsFromRawSqlQueryFactory.php (hunk 2 above)
Comment (homersimpsons) [reply src/QueryFactory/FindObjectsFromRawSqlQueryFactory.php:133]: In fact didn't notice, thanks --'

File: src/QueryFactory/FindObjectsFromRawSqlQueryFactory.php (hunk 3 above)
Comment (mhtghn) [reply src/QueryFactory/FindObjectsFromRawSqlQueryFactory.php]: done

**Review by homersimpsons — APPROVED** [review]

**Pull request merged**